front/pages/api/w/[wId]/data_sources/request-access.ts

```diff
@@ -0,0 +1,96 @@
+import { isLeft } from "fp-ts/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+import sanitizeHtml from "sanitize-html";
+
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { sendEmail } from "@app/lib/email";
+import { apiError } from "@app/logger/withlogging";
+
+export const PostRequestAccessBodySchema = t.type({
+  email: t.string,
+  emailMessage: t.string,
+  emailRequester: t.string,
```

Comment:
Should we expose this endpoint behind `/data_sources/[name]/request-access` so we can infer here the email address of the owner of the data source/data source view?

Reply:
I think leaving it here would enable us to have a general request-access endpoint. There could be a world in which users ask to an admin to set up a new connector 🤷🏼‍♂️